ds3_java_cli/src/main/java/com/spectralogic/ds3cli/Main.java

```diff
@@ -27,29 +36,90 @@
         LOG.setLevel(Level.ALL);
 
-        /*
-        final Appender fileAppender =  LOG.getAppender("LOGFILE");
-        final ch.qos.logback.classic.filter.ThresholdFilter fileFilter = new ThresholdFilter();
-        fileFilter.setLevel(consoleLevel);
-        fileFilter.setName(consoleLevel);
-        fileAppender.addFilter(fileFilter);
-        fileFilter.start();
-        */
-
-        final Appender consoleAppender =  LOG.getAppender("STDOUT");
-        final ch.qos.logback.classic.filter.ThresholdFilter consoleFilter = new ThresholdFilter();
-        consoleFilter.setLevel(fileLevel);
-        consoleFilter.setName(fileLevel);
-        consoleAppender.addFilter(consoleFilter);
-        consoleFilter.start();
+        if (!consoleLevel.equalsIgnoreCase(Level.OFF.toString())) {
+            // create and add console appender
+            final PatternLayoutEncoder consoleEncoder = new PatternLayoutEncoder();
+            consoleEncoder.setContext(loggerContext);
+            consoleEncoder.setPattern(LOG_FORMAT_PATTERN);
+            consoleEncoder.start();
+
+            final ConsoleAppender<ILoggingEvent> consoleAppender = new ConsoleAppender<ILoggingEvent>();
+            consoleAppender.setContext(loggerContext);
+            consoleAppender.setName("STDOUT");
+            consoleAppender.setEncoder(consoleEncoder);
+
+            final ch.qos.logback.classic.filter.ThresholdFilter consoleFilter = new ThresholdFilter();
+            consoleFilter.setLevel(consoleLevel);
+            consoleFilter.setName(consoleLevel);
+            consoleFilter.start();
+            consoleAppender.addFilter(consoleFilter);
+
+            consoleAppender.start();
+            LOG.addAppender(consoleAppender);
+        }
+
+        if (!fileLevel.equalsIgnoreCase(Level.OFF.toString())) {
+            // create file appender only if needed.
+            // if done in the xml, it will create an empty file
+
+            final RollingFileAppender<ILoggingEvent> fileAppender = new RollingFileAppender<ILoggingEvent>();
+            final TimeBasedRollingPolicy<ILoggingEvent> timeBasedRollingPolicy = new TimeBasedRollingPolicy<ILoggingEvent>();
+            final DefaultTimeBasedFileNamingAndTriggeringPolicy<ILoggingEvent> timeBasedTriggeringPolicy = new DefaultTimeBasedFileNamingAndTriggeringPolicy<ILoggingEvent>();
+
+            fileAppender.setContext(loggerContext);
+            timeBasedTriggeringPolicy.setContext(loggerContext);
+            timeBasedRollingPolicy.setContext(loggerContext);
+
+            fileAppender.setRollingPolicy(timeBasedRollingPolicy);
+            timeBasedRollingPolicy.setParent(fileAppender);
+            timeBasedRollingPolicy.setTimeBasedFileNamingAndTriggeringPolicy(timeBasedTriggeringPolicy);
+            timeBasedTriggeringPolicy.setTimeBasedRollingPolicy(timeBasedRollingPolicy);
+
+            timeBasedRollingPolicy.setFileNamePattern(LOG_DIR + LOG_ARCHIVE_FILE_PATTERN);
```

Comment:
My infusion was rescheduled for today. I am on the hospital WiFi and can’t get VPN or HipChat. You can reach me by email.

I pushed a version with time based rolling and your other fixes.
jk

From: Ryan Moore [mailto:notifications@github.com]
Sent: Tuesday, July 19, 2016 11:11 PM
To: SpectraLogic/ds3_java_cli <ds3_java_cli@noreply.github.com>
Cc: John Kranz <johnk@spectralogic.com>; Author <author@noreply.github.com>
Subject: Re: [SpectraLogic/ds3_java_cli] JAVACLI-79 Instantiate logback config only as needed. (#104)


In ds3_java_cli/src/main/java/com/spectralogic/ds3cli/Main.java<https://github.com/SpectraLogic/ds3_java_cli/pull/104#discussion_r71466173>:

> +            // if done in the xml, it will create an empty file

> +

> +            final RollingFileAppender<ILoggingEvent> fileAppender = new RollingFileAppender<ILoggingEvent>();

> +            final TimeBasedRollingPolicy<ILoggingEvent> timeBasedRollingPolicy = new TimeBasedRollingPolicy<ILoggingEvent>();

> +            final DefaultTimeBasedFileNamingAndTriggeringPolicy<ILoggingEvent> timeBasedTriggeringPolicy = new DefaultTimeBasedFileNamingAndTriggeringPolicy<ILoggingEvent>();

> +

> +            fileAppender.setContext(loggerContext);

> +            timeBasedTriggeringPolicy.setContext(loggerContext);

> +            timeBasedRollingPolicy.setContext(loggerContext);

> +

> +            fileAppender.setRollingPolicy(timeBasedRollingPolicy);

> +            timeBasedRollingPolicy.setParent(fileAppender);

> +            timeBasedRollingPolicy.setTimeBasedFileNamingAndTriggeringPolicy(timeBasedTriggeringPolicy);

> +            timeBasedTriggeringPolicy.setTimeBasedRollingPolicy(timeBasedRollingPolicy);

> +

> +            timeBasedRollingPolicy.setFileNamePattern(LOG_DIR + LOG_ARCHIVE_FILE_PATTERN);

We shouldn't use a time based rolling policy, but a sized based one instead.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/SpectraLogic/ds3_java_cli/pull/104/files/24df5e3f417d5483d4fc684e901d165d9bc747b3#r71466173>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ANGH76rcM2w4IFm_12ggRMS76VziMTJjks5qXa3JgaJpZM4JQHr2>.